src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -212,6 +270,29 @@ private void sendRequest(
       conn.writeToChannel(request);
     }
 
+    private void handleChunk(Buffer buff) {
+      HttpClientResponseImpl r;
+      synchronized (conn) {
+        if (paused) {
+          if (pausedChunk == null) {
+            pausedChunk = buff.copy();
+          } else {
+            pausedChunk.appendBuffer(buff);
+          }
+          return;
+        } else {
+          if (pausedChunk != null) {
```

Comment:
Just for my understanding:
This part is only here to support the multithreaded case.
In the case of regular EventLoop usage the doResume should have taken care of everything before handleChunk could be hit again.

Reply:
actually it is necessary because there is a possible race condition since the actual resuming uses `runOnContext` to deliver the chunk asynchronously (`doResume` might be called from non vertx thread): so it's possible to have `paused == false` and `pausedChunk != null`.

This part is actually going away with a PR I'm currently working on (https://github.com/vert-x3/wiki/wiki/Vert.x-Roadmap#stream-improvements). The idea of this PR is to prepare the http client connection so the stream behaviour is replaced by the next PR (and also in all other places of vertx).

Reply:
In my working branch the new implementation is simply:

```
    private boolean handleChunk(Buffer buff) {
      return queue.push(buff);
    }
```

Reply:
Ah crap, yes, overlooked the runOnContext.

Reply:
that's why we want to encapsulate such racy prone code in a single class :-)